api/middleware/auth.js

```diff
@@ -36,6 +37,27 @@ const can = activity =>
       });
       // First check if they're logged in
       module.exports.loggedIn(req, res, () => {
+
+        if (Array.isArray(activity) ){
+          const hasActivity = activity.some((action) => {
```

Comment:
That was not my mental model.  I was envisioning something like ['view-document', 'edit-document'] on an endpoint for fetching an APD or ['view-roles', 'edit-roles'] for fetching a list of people and their roles.  If you needed 2 or more permissions to do something I think that should probably be codified into another kind of permission like 'create-document'.  I want to make these as clear and reusable as possible while also keeping the number of different kinds of permissions low to prevent confusion.

Reply:
So you were running into a situation where a user had like edit-roles, but not view-roles? I feel like that would be better handled by fixing the roles. I'm more of the rule of least privilege mind. I would think if multiple there were multiple cans you would need to be able to do all of them. Thoughts? @tbolt can you weigh in too?

Reply:
Could we add something to specify to can "all" or "any" just so it's easy to tell what the code is trying to do? Maybe add another arg?

Reply:
This is a tough one! I think I lean more towards `.every` and we should be explicit in what we pass to this check. Though I could see `.some` being more suitable to our environment since we could pass the users entire activity list in. Doing `.every` is safer though since it would only be annoying, whereas misuse with `.some` could be a vulnerability.

Reply:
ok.  I will change it and redo the tests a little bit to match the new logic.

Reply:
👍 thanks Knoll

Reply:
@thetif @tbolt this is now .every and tests have been updated accordingly.